pyEpiabm/pyEpiabm/tests/test_sweep/test_host_progression_sweep.py

```diff
@@ -48,31 +68,37 @@ def test_update_time(self):
         self.assertTrue(1 <= time <= 10)
 
     def test_call(self):
-        """Test the main function of the Host Progression Sweep.
-        Person 2 is set to exposed and becoming infectious in one time step.
-        Checks the populations updates as expected.
+        """Tests the main function of the Host Progression Sweep.
+        Person 3 is set to susceptible and becoming exposed. Person 2 is set to
+        exposed and becoming infectious in one time step. Checks the
+        population updates as expected.
         """
         self.person2.infection_status = pe.property.InfectionStatus.Exposed
         self.person2.time_of_status_change = 1
         self.person2.next_infection_status = \
             pe.property.InfectionStatus.InfectMild
+        self.person3.infection_status = pe.property.InfectionStatus.Susceptible
+        self.person3.time_of_status_change = 1
+        self.person3.next_infection_status = \
+            pe.property.InfectionStatus.Exposed
         test_sweep = pe.sweep.HostProgressionSweep()
         test_sweep.bind_population(self.test_population)
 
         # Tests population bound successfully.
         self.assertEqual(test_sweep._population.cells[0].persons[1].
                          infection_status, pe.property.InfectionStatus.Exposed)
 
-        # Test that the call fails for non-int time.
-        self.assertRaises(TypeError, test_sweep, 0.5)
-
```

Comment:
Are we certain it can handle non-integer time if we are removing this?

Reply:
The swap of if for while time of update status change less than or equal to simulation time fixes this I think.